README
################################################################################
#
# OCS Inventory NG Management Server Setup
#
# Copyleft 2006 Pascal DANEK
# Web: http://www.ocsinventory-ng.org
#
# This code is open source and may be copied and modified as long as the source
# code is always made freely available.
# Please refer to the General Public Licence http://www.gnu.org/ or Licence.txt
################################################################################

How to setup OCS Inventory NG Management Server ?

================================================================================
CAUTION: 
If upgrading Communication server from OCS Inventory NG 1.0 RC2 and
previous, please remove any Apache configuration for Communication Server !

Setup now install files in accordance with Filesystem Hierarchy Standard. So,
no file is installed under Apache root document directory (Refer to Apache
configuration files to locate it).
If you're upgrading from OCS Inventory NG Server 1.01 and previous, YOU MUST
REMOVE (or move) directories 'ocsreports' and 'download' from Apache root
document directory.
If you choose to move directory, YOU MUST MOVE 'download' directory to 
Administration Server variable/cache directory (by default 
/var/lib/ocsinventory-reports), especially if you use deployment feature.
================================================================================


OCS Inventory NG Management Server needs the following components:
- MySQL version 4.1 or higher
- PERL 5.6 or higher (5.8 or higher recommended)
- Apache version 1.3.33 or higher / Apache version 2.0.X or higher
- Apache Mod_perl version 1.29 or higher
- PHP 4.3.2 or higher, with ZIP support enabled.
- Apache Mod_php version 4.3.2 or higher
- Perl module XML::Simple version 2.12 or higher
- Perl module Compress::Zlib version 1.33 or higher
- Perl module DBI version 1.40 or higher
- Perl module DBD::Mysql version 2.9004 or higher
- Perl module Apache::DBI version 0.93 or higher
- Perl module Net::IP version 1.21 or higher
- Perl module SOAP::Lite version 0.66 or higher, not required, used only with
  SOAP web service
- Perl module XML::Entities version 0.02 or higher, not required, used only with
  SOAP web service

You must also know the following things (setup will try to found them, but
no garanty):
- Host running MySQL server (generally localhost)
- Port of MySQL server (generally 3306)
- Path to Apache web server binary file (usually named httpd or apache or
  apache2)
- Path to Apache main configuration file (usually named httpd.conf or
  apache.conf or apache2.conf)
- Path to Apache configuration directory (usually named conf.d)
- User account running Apache web server (usually apache or www-data)
- Group of user account running Apache web server (usually the same as user
  account)
- Version of Apache mod_perl
- Where to put Communication server log directory (usually
  /var/log/ocsinventory-server)

Run "sh setup.sh" and follow the instructions. It will ask you if you want to
setup Communication Server and/or Administration Server, and will do all
configuration for you.

Using default setup,
- Communication Server is installed as standard PERL module
- Communication Server configuration file ocsinventory-server.conf is installed
  under Apache configuration directory, usually /etc/httpd/conf.d or
  /etc/apache[2]/conf.d
- Communication Server logs are stored under directory 
  /var/log/ocsinventory-server
- PHP files are installed under directory 
  /usr/share/ocsinventory-reports/ocsreports
- Writable/cache directories "download" and "ipd" are created under directory 
  /var/lib/ocsinventory-reports
- Administration Server configuration file ocsinventory-reports.conf is 
  installed under Apache configuration directory, usually /etc/httpd/conf.d or
  /etc/apache[2]/conf.d. It contains aliases for /ocsreports and /download URL.
  

Enjoy OCS Inventory NG ;-)

OCS Inventory Team


Apache/Changes
Revision history for Apache::Ocsinventory

2.2 RC
		* Add plugins web service which is used by the plugin engine to install plugin ( Use SOAP::Lite )
		* Add compatibility with Apache 2.4 and more : https://github.com/OCSInventory-NG/OCSInventory-Server/commit/861dcb17f826d9580f0f5f0937bd1ae7376e6602
		
2.2beta1
        * Add <EXCLUDE_TAG> and <EXCLUDE_ID> requests tags in SOAP engine to be able to exclude tags or ids during SOAP requests
        * Add new 'batteries' inventory section to get various batteries data
        * New OCS_OPT_SNMP_PRINT_HTTPS_ERROR configuration option to enable or disable error message display about agent https communication in logfile
        * Fix bug in SOAP engine while retrieving data from type tables

2.1.2

2.1.1
        * Fix Bug #1315004 about MySQL replication broken using MySQL 5.6
        * Fix Bug #1311852 and Bug #1296633 about CHECKSUM updated for all computers
        * Fix a mistake for OCS_PLUGINS_PERL_DIR and OCS_PLUGINS_CONF_DIR options in ocsinventory-server.conf

2.1
        * Add new fields for printers table in Map.pm
        * Fix Bug #1214436 about removing already registered hosts and add Mac support in ipdiscover-util.pl
        * Add a new table for SNMP virtual machines data in Map.pm
        * Fix Bug #1248582 about custom field names not decoded in SOAP ACCOUNTINFO
        * Fix Bug #1191791 about CHECKSUM value not updated when NOTIFY mode send new network changes
        * New plugins system to be able to add new entries in Map.pm and more...

2.1rc1
        * Add availibility to specify a post command for a package deployment
        * SOAP engine is now able to get SNMP devices inventory
        * Add new 'sims' inventory section to get various SIM card data from phone devices	
        * Add availibility to schedule a package deployment
        * Remove OCS_OPT_SUPPORT configuration option and <SUPPORT_LOG> XML tag in PROLOG 
        * Add availibility to reforce a package deployment
        * Add availability to specify custom MIBS for SNMP scans
        * Add availability to specify a custom network address for SNMP scans 
        * New OCS_OPT_DOWNLOAD_EXECUTION_TIMEOUT configuration option to prevent package commands to run indefinitely
        * setup.sh script can now detect Suse/OpenSuse distributions      
        * Add new 'cpus' inventory section to get various CPU data	
        * Add a new field in hardware section to get the computer architecture	
        * Add new fields for snmp_loadbalancer table in Map.pm
        * Fix Bug #1085020 about ERR_ALREADY_IN_HISTORY message that should be success 
        * Prevent bugs in NOTIFY mode with older XML behaviour

2.0.5
        * Create "logs" and "scripts" directories for administration console in setup.sh

2.0.4

2.0.3
        * Fix Bug #881406 about security issue due to snmp_com.txt file which is no longer used
        * Do not take care of comment lines when getting external useragents file content
        * Fix various bugs in NOTIFY mode

2.0.2
        * Add availibilty to allow external useragents using a text file

2.0.1
        * Remove 'Satisfy any' Apache directive in ocsinventory-server.conf
        * Fix RPM packages names in setup.sh
        * Fix Bug #800418 about subnet, gateway and IP DHCP informations disappearance from database
        * Increase php_value post_max_size and php_value upload_max_filesize values to 101 Mo and 100 Mo in ocsinventory-reports.conf
        * Change php_value to php_flag for magic_quotes_gpc setting in ocsinventory-reports.conf
        * Add new useragents for iOS and Android OCS agents in Useragent.pm

2.0
        * New OCS_OPT_SUPPORT configuration option for OCS Inventory support activation
        * Add a new <SUPPORT_LOG> XML tag in PROLOG
        * Add a new module to manage OCS support: Support.pm
        * Add a <DHCP> XML tag in NOTIFY request to update IPDHCP field in database
2.0 rc4
        * Add a warning in setup.sh about what has to be done if the OS has newsyslog installed (BSD systems)               
        * Fix Bug #744022 about logrotate configuration in setup.sh even if the OS has not logrotate installed
        * Add availability to specify mysql socket in ipdiscover-util.pl
        * Fix UTF8 problem while comparing accountinfo strings after inventory step
2.0 rc3
        * Use OCS_OPT_DBI_MYSQL_SOCKET in SOAP engine if needed
        * Ipdiscover now uses useragent instead of OSNAME for automatic election
        * Fix Bug #721278 about hardware_osname cache table. This table is now feeded by engine
        * Verify that HARDWARE_ID are not groups when computing groups caches
        * Disable MySQL strict mode by default
        * Fix Bug #726522 about OCS_OPT_DBI_MYSQL_SOCKET option in ocsinventory-server.conf
        * Don't process the sections that are use for capacities special inventory in SOAP engine (but thats the way its meant to be)
        * Fix a bug in Download.pm: ERR_ALREADY_IN_HISTORY message was never added in devices table
        * Fix SQL syntax error and add a capacities tables exclusion in Duplicate.pm

2.0 rc2
        * Add a new variable $Apache::Ocsinventory::BUILD_VERSION in Ocsinventory.pm
        * Fix Bug #694760 about utf8 strings broken

2.0 rc1
        * New OCS server module : Snmp.pm . It is use for SNMP integration in OCS Inventory NG. This module comes with Snmp/Data.pm and Snmp/Inventory.pm
        * Add Useragent.pm to control agents and their versions to solve versions incompatibilty problems between OCS agents and OCS server
        * Add a new parameter for tables in Map.pm: mandatory. This parameter force database feeding even if inventory section is not in XML
        * Control if inventory section is in XML before writing it in database to avoid bugs.
        * Add COMMENTS and VERSION fields in snmp_softwares table
        * Add new fields in softwares table: GUID, LANGUAGE, LASTDATE and BITSWIDTH

Apache/Makefile.PL
#-------------------------------------------------------
#
# $Id: Makefile.PL,v 1.6 2008-02-27 21:26:27 dliroulet Exp $
#
#-------------------------------------------------------

################################################################################
#
# OCS Inventory NG Communication Server Perl Module Setup
#
# Copyleft 2006 Pascal DANEK
# Web: http://www.ocsinventory-ng.org
#
# This code is open source and may be copied and modified as long as the source
# code is always made freely available.
# Please refer to the General Public Licence http://www.gnu.org/ or Licence.txt
################################################################################

use ExtUtils::MakeMaker;

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

WriteMakefile(
    'NAME'         => 'Apache::Ocsinventory',
    'VERSION_FROM' => 'Ocsinventory.pm',
    'PREREQ_PM'    => {
			'DBI'             => 1.40,   # no DBI can never be any fun.
			'Apache::DBI'     => 0.93,   # no Apache::DBI can never be any fun.
			'Compress::Zlib'  => 1.33,   # no Compress:Zlib can never be any fun.
			'DBD::mysql'      => 2.9004, # no DBD::mysql can never be any fun.
			'XML::Simple'     => 2.12,   # no XML::Simple can never be any fun.
			'Net::IP'         => 1.21,   # no Net::IP can never be any fun.
	},
#    'EXE_FILES'         => ['binutils/ocsinventory-injector.pl'],
#    'INSTALLSCRIPT'     => "/usr/bin",
#    'INSTALLSITESCRIPT' => "/usr/bin",
    ($] >= 5.005 ?      ## Add these new keywords supported since 5.005
      (AUTHOR           => 'Pascal DANEK <hunal@users.sourceforge.net>') : ()),
);

# EOF
